### DISCUSSION TITLE
Allow using a template for more than one host using ENV

### DISCUSSION THREAD

**Discussion by etopian** [nginx-proxy/nginx-proxy#1943] [Q&A]
Is it possible to add a NGINX_TEMPLATE variable to ENV of the target container and use that as a filename for the docker-gen template used for that container? Say I want to start 10 containers, using the same template, and 10 others using a different template. The current procedure to do that is cumbersome, it would be better to be able to supply that as a part of the run argument for the target container and have that just work. Making a template for each individual container or even copying over the file is a bit messy and cumbersome.

**Comment by tkw1536** [comment]
There is [Per-VIRTUAL_HOST default configuration](https://github.com/nginx-proxy/nginx-proxy#per-virtual_host-default-configuration), but this does allow you to use any kind of templating in the configuration file.
